### DISCUSSION TITLE
Goals & Notes

### DISCUSSION THREAD

**Discussion by iurimatias** [status-im/nim-status#171] [Development]
### Goals

Primary goals
* status-go feature parity
* make it work with desktop
* must work with compiler safety checks
* must be async (nim-chronos not asyncdispatch)

Other goals
* make it work status-react
* code should have tests & documentation

### Observations:

* API compatibility with status-go is not a goal/priority
* we need to move fast to catch up to status-go
* we should take into account compatibility of the async APIs
* It should (must?) be compatible with existing status-go sqlite datbase
* single-threaded library, responsibility to spawn threads up to consumer
* exposed api procs are all async

### Approach

To speed up development the library will be development with a very simple command line client, later integrated into desktop/mobile

**Comment by corpetty** [comment]
looking forward to a command line client to play around with.
